src/goto-symex/renaming_level.h

```diff
@@ -84,6 +86,63 @@ struct symex_level2t : public symex_renaming_levelt
 {
   symex_level2t() = default;
   ~symex_level2t() override = default;
+
+  /// In the case of a state split, we want to keep track of the overall
+  /// name generation across all states for identifier creation.
+  optionalt<current_namest> local_generations;
```

Comment:
It looks like you're using the existing `current_names` to track global-highest-generation and this new `local_generations` to track the latest version on a particular path? If so, see #3980 -- that creates `unique_index_map` to store a global highest L1 generation, and that map belongs to `path_storaget`, not `symex_level2t` which is path-local. (as a rule: varies per path -> `goto_statet`, varies per path but doesn't need to be copied on split -> `goto_symex_statet`, doesn't vary per path -> `path_storaget`). So you should have a map `unique_l2_index_map` or something on `path_storaget` used to allocate fresh L2 names, and `current_names` should continue to describe what L2 generation is observed by a read on a particular path.